packages/overlay/src/integrations/sentry/index.ts

```diff
@@ -103,34 +103,51 @@ export default function sentryIntegration(options?: SentryIntegrationOptions) {
   } satisfies Integration<Envelope>;
 }
 
+/**
+ * Implements parser for
+ * @see https://develop.sentry.dev/sdk/envelopes/#serialization-format
+ * @param rawEvent Envelope data
+ * @returns parsed envelope
+ */
 export function processEnvelope(rawEvent: RawEventContext) {
   const { data } = rawEvent;
-  const [rawHeader, ...rawEntries] = data.trim().split(/\n/gm);
-
-  const header = JSON.parse(rawHeader) as Envelope[0];
-  const items: Envelope[1][] = [];
-  for (let i = 0; i < rawEntries.length; i += 2) {
-    // guard both rawEntries[i] and rawEntries[i + 1] are defined and not empty
-    if (!rawEntries[i] || !rawEntries[i + 1]) {
-      continue;
+  let prevCursor = 0;
+  let cursor = data.indexOf('\n');
+  const envelopeHeader = JSON.parse(data.slice(prevCursor, cursor).toString()) as Envelope[0];
```

Comment:
If you get a buffer in, shouldn't you specify encoding when converting to string?